lib/ansible/modules/windows/win_certificate_info.ps1

```diff
@@ -0,0 +1,145 @@
+    $cert_info.signature_algorithm = $cert.SignatureAlgorithm.FriendlyName
+    $cert_info.dns_names = @($cert_info.issued_to)
+    $cert_info.raw = [System.Convert]::ToBase64String($cert.GetRawCertData())
+    $cert_info.public_key = [System.Convert]::ToBase64String($cert.GetPublicKey())
+    $cert_info.extensions = @()
+    foreach ($extension in $cert.Extensions)
+    {
+        $cert_info.extensions += @{
+            critical = $extension.Critical
+            field    = $extension.Oid.FriendlyName
+            value    = $extension.Format($false)
+        }
+        if ($extension -is [System.Security.Cryptography.X509Certificates.X509BasicConstraintsExtension])
+        {
+            $cert_info.is_ca = $extension.CertificateAuthority
+            $cert_info.path_length_constraint = $extension.PathLengthConstraint
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509EnhancedKeyUsageExtension])
+        {
+            $cert_info.intended_purposes = $extension.EnhancedKeyUsages.FriendlyName -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509KeyUsageExtension])
+        {
+            $cert_info.key_usages = $extension.KeyUsages.ToString().Split(',').Trim() -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension])
+        {
+            $cert_info.ski = $extension.SubjectKeyIdentifier
+        }
+        elseif ($extension.Oid.value -eq '2.5.29.17')
+        {
+            $sans = $extension.Format($true).Split("`r`n", [System.StringSplitOptions]::RemoveEmptyEntries)
+            foreach ($san in $sans)
+            {
+                $san_parts = $san.Split("=")
+                if ($san_parts.Length -ge 2 -and $san_parts[0].Trim() -eq 'DNS Name')
+                {
+                    $cert_info.dns_names += $san_parts[1].Trim()
+                }
+            }
+        }
+    }
+    return $cert_info
+}
+
+$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues() | ForEach-Object { $_.ToString() }
+
+$spec = @{
+    options = @{
+        thumbprint     = @{ type = "str"; required = $false }
+        store_name     = @{ type = "str"; default = "My"; }
+        store_location = @{ type = "str"; default = "LocalMachine"; choices = $store_location_values; }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$thumbprint = $module.Params.thumbprint
+$store_name = $module.Params.store_name
+$store_location = [System.Security.Cryptography.X509Certificates.Storelocation]"$($module.Params.store_location)"
+
+$store = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Store -ArgumentList $store_name, $store_location
+try
+{
+    $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+}
+catch [System.Security.Cryptography.CryptographicException]
```

Comment:
I think I went the wrong route with `win_certificate_store` in trying to catch every exception type that was raised in some of the .NET calls. It was from an era when Ansible's messages for uncaught exceptions were quite minimal and sometimes cryptic.

Since 2.8 we properly format any uncaught exceptions to include things like the stack trace and actual PowerShell message making it easy and simple for people to see:

1. Where the exception was raised,
2. What the exception type was, and
3. The actual line where the exception was raised

This coupled with the fact that we can reduce the lines from 16 to just 1 makes the code a lot simpler. The only times I think a try/catch should be used today is when

1. We can actually recover from the exception,
2. Using it for flow control, to determine the next path, or
3. The exception message returned is quite vague and we need to add some context around it that may not be apparent by default